aip/firebase/3260.md

```diff
@@ -0,0 +1,46 @@
+---
+aip:
+  id: 3260
+  scope: firebase
+  state: draft
+  created: 2020-06-10
+  updated: 2019-06-10
+permalink: /firebase/3260
+redirect_from:
+  - /3260
+---
+
+# Firebase CLI
+
+## Guidance
+
+### Project Selection
+
+The Firebase CLI works with "project directories" that can be associated with
+one or more Firebase projects via the `use` command. This ambient project
+selection can be overridden by passing the `-P` or `--project` flag.
+
+All Firebase CLI commands **must** determine the intended project for a command
+based on common project selection, except when a command involves multiple
+projects (e.g. a command that clones configuration from one project to
+another).
+
+### Arguments and Flags
+
+All flags passed to the CLI **must** be optional. Required inputs for a command
+should be kept to a minimum, and when necessary **must** be arguments.
+
+Flags **must** be named using kebab case (e.g. `--example-flag`) and **may**
+provide a single-letter alias that does not conflict with common global flags.
+
+### Naming Commands
+
+- Commands **must** use parameters for arguments.
+- Commands **must** use a `:list` for listing multiple resources.
+
+### Deployment
+
+Behavior that takes local state (e.g. a rules file) and applies it to the live
+operation of a Firebase project **must** integrate with the `deploy` command.
```

Comment:
For commands like `firebase ext:update <ext-instance-id> <localDirectory>` or some of the new Hosting cmds that deploy to the live site... are those compliant with this guidance?  I guess I'm not clear on what "integrate" means?

Reply:
The guidance is intended to make sure that if a product is doing "deploy-like" things it integrates with the common `firebase deploy` command.

Hosting is integrated with `firebase deploy`, we're just exploring some additional ways to interact with the project that will be surfaced in other commands.

Extensions is nuanced, but TL;DR we will eventually bring Extensions to `firebase deploy` as well (public repo so details left to imagination) 😄

All that said can probably be more clear here.